### DISCUSSION TITLE
Workflow isn't enabled in repos generated from template

### DISCUSSION THREAD

**Discussion by vromash** [community/community#25659] [Actions]
<p>Hello!</p>
<p><strong>N.B.</strong> This issue appeared today morning. I had working flow for more than a week.</p>
<p><strong>The issue:</strong><br>
I have a template repository with working workflow in my organization. When I create new repo from this template, workflows aren’t enabled automatically.</p>
<p>In Actions section I can see broken stylings:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/d/7db2b56ce2f79f86fb75fb9e74d6bb0d41bc1cf7.png" data-download-href="/uploads/short-url/hVYz3s6b0ItzbPUFrqSjV66Et1B.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/26174786/181092392-42eea2ba-02b6-4907-b361-6a6b02f44dd4.png" alt="image" data-base62-sha1="hVYz3s6b0ItzbPUFrqSjV66Et1B" width="690" height="186" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/d/7db2b56ce2f79f86fb75fb9e74d6bb0d41bc1cf7_2_690x186.png, https://aws1.discourse-cdn.com/github/optimized/3X/7/d/7db2b56ce2f79f86fb75fb9e74d6bb0d41bc1cf7_2_1035x279.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/7/d/7db2b56ce2f79f86fb75fb9e74d6bb0d41bc1cf7_2_1380x372.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/d/7db2b56ce2f79f86fb75fb9e74d6bb0d41bc1cf7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1913×518 31.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>After page reload stylings become even more broken, footer disappears and central div goes to the left.<br>
Both repository and organization settings have <code>Allow all actions</code> tick.</p>
<p>Does anyone now what happened?<br>
Is this behavior expected?<br>
Is there fix or workaround?</p>
<p>Thanks!</p>

**Comment by munderseth** [comment]
<p>We also have a similar use-case that just recently <strong>quit working</strong>. We create a new Repo via API using a repository template. The template repo contains a workflow. In our case, we typically create the new Repo in a 2nd organization, which can be done using the API.</p>
<p>I do not believe there is any API that can enable the disabled workflows, for the case <code>Workflows aren't being run on this repository</code>.</p>
<p>Note that I understand the reason for <code>disabling</code> workflows for a <em>forked</em> Repo.</p>

**Comment by dmusican** [comment]
<p>I’m having the same problem, using GitHub Classroom which makes heavy use of template repositories. All of my students are finding that their tests are now not running, unless they enable actions manually as in the above instructions.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/vromash">@vromash</a>, <a class="mention" href="/munderseth">@munderseth</a>, <a class="mention" href="/dmusican">@dmusican</a>,</p>
<p>Sometimes, when creating a new repository, you need to manually enable the Actions on the repository via clicking the “<strong>Enable Actions on this repository</strong>” button. Most of time, this situation occurs when you fork a repository and creating a new repository from template.</p>
<br>
<blockquote>
<p>Both repository and organization settings have <strong>Allow all actions</strong> tick.</p>
</blockquote>
<p>This option is not related this question. The options “<strong>Allow all actions</strong>”, “<strong>Allow local actions only</strong>” and “<strong>Allow select actions</strong>” are used to set which actions are allowed to be used in the workflows within current organization or repository.</p>
<ul>
<li>
<p><strong>Allow all actions</strong> - - actions from anywhere are allowed to be used in the workflows within current organization or repository.</p>
</li>
<li>
<p><strong>Allow local actions only</strong> - - actions only defined in the repositories within current organization are allowed to be used in the workflows within current organization or repository.</p>
</li>
<li>
<p><strong>Allow select actions</strong> - - actions only match the specified criteria are allowed to be used in the workflows within current organization or repository.</p>
</li>
</ul>
<p>To view more details, you can see “<a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-organizations-and-teams/disabling-or-limiting-github-actions-for-your-organization" rel="noopener nofollow ugc"><strong>Disabling or limiting GitHub Actions for your organization</strong></a>”.</p>

**Comment by hsz** [comment]
<aside class="quote group-github-partners" data-username="brightran" data-post="4" data-topic="136421">
<div class="title">
<div class="quote-controls"></div>
 brightran:</div>
<blockquote>
<p>Sometimes, when creating a new repository, you need to manually enable the Actions on the repository via clicking the “ <strong>Enable Actions on this repository</strong> ” button.</p>
</blockquote>
</aside>
<p><a class="mention" href="/brightran">@brightran</a>, before that issue was reported, enabling Actions was not required at all for the Template Repositories at all.</p>
<p>I have created and maintain <a href="https://github.com/JetBrains/intellij-platform-plugin-template" rel="noopener nofollow ugc">IntelliJ Platform Plugin Template</a> project that is supposed to run the <em>Template Cleanup</em> workflow when the user clicks <em>Use this template</em> button - i.e., to set the plugin project name to match the current repository name, cleanup README, etc.<br>
In general, it’s a part of the workflow that is required to make everything working.</p>
<p>For now, there are over 400 projects created from that template with that workflow successfully triggered.</p>
<p>I find the current behavior as a bug and requiring users to enable Actions and pushing a commit to the remote repository breaks the whole concept of that template.</p>
<p>Is there anything we can do about that?</p>

**Comment by dmusican** [comment]
<p><a class="mention" href="/brightran">@brightran</a> Can you clarify what you mean by “sometimes”? Again, this is a change in behavior from what it was just a couple of days ago. This is affecting teachers who use <a href="https://education.github.community/t/automated-tests-arent-running-and-how-to-start-them-again/63095/4">GitHub Classroom</a> in a way it didn’t just two days ago, and makes the experience for students more difficult.</p>

**Comment by vromash** [comment]
<p><a class="mention" href="/brightran">@brightran</a><br>
As mentioned above, workflows were enabling automatically at the start of the week. Our project is built around actions and repo generating from template. While it’s not working we can’t continue development.</p>
<p>It would be great if the old functionality could be returned back or created endpoint which allowed us to enable workflows.</p>

**Comment by chrispat** [comment]
<p>When you click on the Actions tab do you see a button to enable actions on the repo?</p>

**Comment by hsz** [comment]
<p>Yes, because now they’re disabled by default - which is not something that we’d expect in repositories created from templates.<br>
Previously, Actions were enabled - which made the template-based project fully operational.</p>

**Comment by mine-cetinkaya-rundel** [comment]
<p>I’ve raised this issue in the GitHub Education forum as well (see <a href="https://education.github.community/t/enabling-actions-on-repos-created-from-templates/63167/3">here</a> for the thread). Like <a class="mention" href="/dmusican">@dmusican</a> my use case is education-focused and this change broke our workflow for providing automated feedback to students. It might seem trivial to ask students to push a button, but the workflow is designed around students not having to do that (and add to the cognitive load). I hope GitHub will consider reverting this change, especially considering we’re in the middle of one of the hardest semesters of our lives and are not in a position to help students in person.</p>

**Comment by akosdudas** [comment]
<p>Same experience with my classes. The Actions tab shows the state as disabled and has the button to enable. But at the same time going to the repository settings (as owner of the repository) and checking the Actions settings it does not say that the feature is disabled, it says “Actions permissions: Allow all actions”.</p>
<p>I also checked enabling this via API, but there is no such functionality.</p>

**Comment by hamelsmu** [comment, marked as answer]
<p>Hello everyone, this issue has now been resolved.  Actions are enabled on template repos now by default.  It was a bug that was fixed that was fixed just now and has been pushed to production.</p>

**Comment by hsz** [comment]
<p>That’s great news! Thank you for sharing it, <a class="mention" href="/hamelsmu">@hamelsmu</a>!<br>
I can confirm that Actions are enabled by default now in the project created from the template. :muscle:</p>

**Comment by munderseth** [comment]
<p>Thanks. I have <strong>confirmed that our use-case works now</strong> - Repo creation using API with template “across” organizations. :+1:</p>

**Comment by macpham** [comment]
<p>Is this still working for people?</p>

**Comment by Siam-Rayhan1** [comment]
Every time i click "Enable Action on the workflow"
I got  ``` Unable to enable Actions for this repository.``` 
How do i fix it?

**Comment by strainovic** [comment]
This still does not work. Any update? Thanks

**Reply by strainovic** [reply]
It looks like it works now. It doesn't matter as long as it works :)
